pkg/controller/flowsconfig/flowsconfig_controller.go

```diff
@@ -0,0 +1,135 @@
+package flowsconfig
+
+import (
+	"context"
+	"log"
+	"strconv"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/openshift/cluster-network-operator/pkg/names"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	ovsFlowsConfigMapName   = "ovs-flows-config"
+	ovsFlowsConfigNamespace = names.APPLIED_NAMESPACE
+)
+
+type FlowsConfig struct {
+	// Target IP:port of the flow collector
+	Target string
+
+	// CacheActiveTimeout is the max period during which the reporter will aggregate flows before sending
+	CacheActiveTimeout *uint
+
+	// CacheMaxFlows is the max number of flows in an aggregate; when reached, the reporter sends the flows
+	CacheMaxFlows *uint
+
+	// Sampling is the sampling rate on the reporter. 100 means one flow on 100 is sent. 0 means disabled.
+	Sampling *uint
+}
+
+// WatchForConfigMap setups the passed controller to watch for any change in the
+// configmap openshift-network-operator/ovs-flows-config
+func WatchForConfigMap(c controller.Controller) error {
+	return c.Watch(&source.Kind{Type: &corev1.ConfigMap{}},
+		handler.EnqueueRequestsFromMapFunc(ReconcileRequests),
+		predicate.ResourceVersionChangedPredicate{},
+	)
+}
+
+// ReconcileRequests filters non-ovs-flows-config events and forwards a request to the
+// openshift-network-operator/cluster operator
+func ReconcileRequests(object client.Object) []reconcile.Request {
+	if object == nil {
+		log.Println(ovsFlowsConfigMapName + ": can't create a reconcile request for a nil object")
+		return nil
+	}
+	n := object.GetName()
+	ns := object.GetNamespace()
+	if n != ovsFlowsConfigMapName || ns != ovsFlowsConfigNamespace {
+		return nil
+	}
+	log.Println(ovsFlowsConfigMapName + ": enqueuing operator reconcile request from configmap")
+	return []reconcile.Request{{NamespacedName: types.NamespacedName{
+		Name:      names.OPERATOR_CONFIG,
+		Namespace: names.APPLIED_NAMESPACE,
+	}}}
+}
+
+// Bootstrap looks for the openshift-network-operator/ovs-flows-config configmap, and
```

Comment:
Shouldn't we default to _something_ for these values if the configmap does not exist?  ((Especially during bootstrap since the configmap most likely won't be there yet if the NOO hasn't started)

Reply:
"Bootstrap" can be executed after any reconcile event and not just on startup.

Reply:
As explained in the comment before, if the configmap does not exist (which will be the usual case), then it will just proceed as now: the `ovnkube-nodes` will be configured normally without any flow tracing capability.